pennylane/capture/autograph/ag_primitives.py

```diff
@@ -84,6 +92,77 @@ def functional_cond():
     set_state(results)
 
 
+def assert_iteration_inputs(inputs, symbol_names):
+    """All loop carried values, variables that are updated each iteration or accessed after the
+    loop terminates, need to be initialized prior to entering the loop.
+
+    The reason is two-fold:
+      - the type information from those variables is required for tracing
+      - we want to avoid accessing a variable that is uninitialized, or uninitialized in a subset
+        of execution paths
+
+    Additionally, these types need to be valid JAX types.
+    """
```

Comment:
I think this is a good point, but I would argue that leaving it public is an oversight, and rather than giving it a docstring we should make it private. There's no reason a user should ever use this, it's purely internal for autograph functionality.